kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/ExecutorExecutionContext.kt

```diff
@@ -42,7 +44,10 @@ object ExecutorExecutionContext : TimeoutExecutionContext {
          scheduler.shutdown()
 
          try {
-            runBlocking {
+            // we use the context from the caller, in order to allow context params to propogate down
+            // into the test case from test extensions
+            // see https://github.com/kotest/kotest/issues/1725
+            runBlocking(context) {
```

Comment:
As mentioned in the original issue description, we should remove the `ContinuationInterceptor` context key here, which contains the `CoroutineDispatcher` element.

```kotlin
runBlocking(context.minusKey(ContinuationInterceptor)) {
   // ...
}
```

According to the documentation of `runBlocking` if we give it a context that includes a `CoroutineDispatcher`, the coroutine will continue to run on the given dispatcher, and the `runBlocking` will just wait for it to finish. Since we want to be able to interrupt this thread, we would just interrupt the waiting but not the actual execution of the test.

Reply:
Thanks, that's what I was missing.